store/store_test.go

```diff
@@ -653,38 +653,75 @@ const MockSearchJSON = `{
 }
 `
 
-func (t *remoteRepoTestSuite) TestUbuntuStoreFind(c *C) {
+func (t *remoteRepoTestSuite) TestUbuntuStoreFindQueries(c *C) {
+	n := 0
 	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		c.Check(r.UserAgent(), Equals, userAgent)
-		q := r.URL.Query()
-		c.Check(q.Get("q"), Equals, "hello")
-		c.Check(q.Get("fields"), Equals, strings.Join(detailFields, ","))
-		w.Header().Set("Content-Type", "application/hal+json")
-		io.WriteString(w, MockSearchJSON)
+		query := r.URL.Query()
+
+		name := query.Get("name")
+		q := query.Get("q")
+
+		switch n {
+		case 0, 1:
+			c.Check(r.URL.Path, Equals, "/search")
+			c.Check(name, Equals, "hello")
+			c.Check(q, Equals, "")
+		case 2:
+			c.Check(r.URL.Path, Equals, "/details/hello")
+			c.Check(name, Equals, "")
+			c.Check(q, Equals, "")
+		case 3:
+			c.Check(r.URL.Path, Equals, "/search")
+			c.Check(name, Equals, "")
+			c.Check(q, Equals, "hello")
+		default:
+			c.Fatalf("what? %d", n)
+		}
+
+		n++
 	}))
 	c.Assert(mockServer, NotNil)
 	defer mockServer.Close()
 
-	var err error
-	searchURI, err := url.Parse(mockServer.URL)
-	c.Assert(err, IsNil)
+	serverURL, _ := url.Parse(mockServer.URL)
+	searchURI, _ := serverURL.Parse("/search")
+	detailsURI, _ := serverURL.Parse("/details/")
 	cfg := SnapUbuntuStoreConfig{
-		SearchURI: searchURI,
+		DetailsURI: detailsURI,
+		SearchURI:  searchURI,
 	}
 	repo := NewUbuntuStoreSnapRepository(&cfg, "")
 	c.Assert(repo, NotNil)
 
-	snaps, err := repo.Find("hello", "", nil)
-	c.Assert(err, IsNil)
-	c.Assert(snaps, HasLen, 1)
-	c.Check(snaps[0].Name(), Equals, "hello-world")
-	c.Check(snaps[0].Prices, DeepEquals, map[string]float64{"EUR": 2.99, "USD": 3.49})
-	c.Check(snaps[0].MustBuy, Equals, true)
+	for _, query := range []string{
+		"hello",
+		"name:hello*",
+		"name:hello",
+		"text:hello",
+	} {
+		repo.Find(query, "", nil)
+	}
+
+	_, err := repo.Find("", "", nil)
```

Comment:
Given that this (and all the following tests) test the bad behaviour, maybe split this part into `TestUbuntuStoreFindBad` (or something along these lines)  ?